cmd/dep/root_analyzer.go

```diff
@@ -167,9 +168,23 @@ func (a *rootAnalyzer) DeriveManifestAndLock(dir string, pr gps.ProjectRoot) (gp
 func (a *rootAnalyzer) FinalizeRootManifestAndLock(m *dep.Manifest, l *dep.Lock, ol dep.Lock) {
 	// Iterate through the new projects in solved lock and add them to manifest
 	// if they are direct deps and log feedback for all the new projects.
+	//
+	// IDEA: Make a lookup map of ProjectRoot ->projects in the old lock
+	// use that data structure to simplify these loops
+	olLookup := make(map[gps.ProjectRoot]gps.LockedProject)
```

Comment:
This currently doesn't identify projects which were removed from the original/imported lock. I see that you used `DiffProjects` from gps and that's a great idea! I hadn't thought of that but it should save us from re-implementing the diff logic.

How about we use `gps.DiffLocks` which will tell us what's been added, removed and modified from the entire locks. Then we don't need to do any looping over the projects, and we can use the resulting `LockDiff` to print feedback on what has been added, and warn about modifies/deletes.